src/components/Table/Table.tsx

```diff
@@ -119,52 +127,215 @@ const Tbody = styled.tbody`
   label: ReactNode;
 }
-type TableRowType = Array<TableCellType>;
-interface TableProps extends HTMLAttributes<HTMLTableElement> {
+interface TableRowType
+  extends Omit<HTMLAttributes<HTMLTableRowElement>, "onSelect" | "id"> {
+  id: string | number;
+  items: Array<TableCellType>;
+}
+
+interface CommonTableProps
+  extends Omit<HTMLAttributes<HTMLTableElement>, "children" | "onSelect"> {
   headers: Array<TableHeaderProps>;
   rows: Array<TableRowType>;
 }
-const Table = forwardRef<HTMLTableElement, TableProps>(
-  ({ headers, rows, ...props }, ref) => (
-    <StyledTable
-      ref={ref}
-      {...props}
+
+type SelectReturnValue = {
+  index: number;
+  item: TableRowType;
+};
+
+interface SelectionType {
+  isSelectable?: boolean;
+  selectedIndices?: Array<number | string>;
+  onSelect?: (indices: Array<SelectReturnValue>) => void;
+}
+
+interface NoSelectionType {
+  isSelectable?: never;
+  selectedIndices?: never;
+  onSelect?: never;
+}
+
+export type TableProps = CommonTableProps & (SelectionType | NoSelectionType);
+
+interface TableBodyRowProps extends Omit<TableRowType, "id"> {
+  headers: Array<TableHeaderProps>;
+  onSelect: (checked: boolean) => void;
+  isSelectable?: boolean;
+  checked: boolean;
+}
+
+const TableBodyRow = ({
+  headers,
+  items,
+  onSelect,
+  isSelectable,
+  checked,
+  ...rowProps
+}: TableBodyRowProps) => {
+  return (
+    <TableRow
+      $isSelectable={isSelectable}
+      {...rowProps}
     >
-      <THead>
-        <tr>
-          {headers.map((headerProps, index) => (
-            <TableHeader
-              key={`table-header-${index}`}
-              {...headerProps}
+      {isSelectable && (
+        <SelectData>
+          <Checkbox
+            checked={checked}
+            onCheckedChange={onSelect}
+          />
+        </SelectData>
+      )}
+      {items.map(({ label, ...cellProps }, cellIndex) => (
+        <TableData
+          key={`table-cell-${cellIndex}`}
+          {...cellProps}
+        >
+          {headers[cellIndex] && <MobileHeader>{headers[cellIndex].label}</MobileHeader>}
+          <span>{label}</span>
+        </TableData>
+      ))}
+    </TableRow>
+  );
+};
+
+const Table = forwardRef<HTMLTableElement, TableProps>(
+  ({ headers, rows, isSelectable, selectedIndices = [], onSelect, ...props }, ref) => {
+    const onSelectAll = (checked: boolean): void => {
+      if (typeof onSelect === "function") {
+        const ids = checked
+          ? rows.map((row, index) => ({
+              item: row,
+              index,
+            }))
+          : [];
+        onSelect(ids);
+      }
+    };
+
+    const onRowSelect =
+      (id: number | string) =>
+      (checked: boolean): void => {
+        if (typeof onSelect == "function") {
+          const selectedItems = rows.flatMap((row, index) => {
+            if (
+              (id === row.id && checked) ||
+              (selectedIndices.includes(id) && id !== row.id)
```

Comment:
this is a bit confusing, why should `selectedIndices` contain an `id` and why do we want the id to be different from the row.id?